src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -78,15 +91,55 @@ class WalletService(
             .filter { it.status == WalletStatusDto.CREATED }
             .switchIfEmpty { Mono.error(WalletConflictStatusException(WalletId(walletId))) }
             .flatMap {
+                ecommercePaymentMethodsClient
+                    .getPaymentMethodById(it.paymentMethodId.value.toString())
+                    .map { paymentMethod -> paymentMethod to it }
+            }
+            .switchIfEmpty {
+                Mono.error(
+                    EcommercePaymentMethodException(
+                        "Payment method id not found",
+                        HttpStatus.NOT_FOUND
+                    )
+                )
+            }
+            .flatMap {
+                val orderId = UUID.randomUUID().toString().replace("-", "").substring(0, 15)
+                val customerId = UUID.randomUUID().toString().replace("-", "").substring(0, 15)
+                val basePath = URI.create(sessionUrlConfig.basePath)
+                val merchantUrl = sessionUrlConfig.basePath
+                val resultUrl = basePath.resolve(sessionUrlConfig.outcomeSuffix)
+                val cancelUrl = basePath.resolve(sessionUrlConfig.cancelSuffix)
+                val notificationUrl =
+                    UriComponentsBuilder.fromHttpUrl(sessionUrlConfig.notificationUrl)
+                        .build(Map.of("orderId", orderId, "paymentMethodId", it.first.id))
+
                 npgClient
                     .createNpgOrderBuild(
                         UUID.randomUUID(),
                         CreateHostedOrderRequest()
-                            .version("2")
-                            .merchantUrl("https://test")
-                            .order(Order())
+                            .version(CREATE_HOSTED_ORDER_REQUEST_VERSION)
+                            .merchantUrl(merchantUrl)
+                            .order(
+                                Order()
+                                    .orderId(orderId)
+                                    .amount(CREATE_HOSTED_ORDER_REQUEST_VERIFY_AMOUNT)
+                                    .currency(CREATE_HOSTED_ORDER_REQUEST_CURRENCY_EUR)
+                                    .customerId(customerId)
+                            )
+                            .paymentSession(
+                                PaymentSession()
+                                    .actionType(ActionType.VERIFY)
+                                    .amount(CREATE_HOSTED_ORDER_REQUEST_VERIFY_AMOUNT)
+                                    .language(CREATE_HOSTED_ORDER_REQUEST_LANGUAGE_ITA)
+                                    .captureType(CaptureType.IMPLICIT)
+                                    .paymentService(it.first.name)
+                                    .resultUrl(resultUrl.toString())
```

Comment:
to add
```
"paymentSession": {
        "actionType": "VERIFY",
        "recurrence": {
            "action": "CONTRACT_CREATION",
            "contractId": "xxx",
            "contractType": "CIT"
        },
```

Reply:
Fixed withe 2f4c96554735a7b18cfed61400a6f10f9b09cad